### PR TITLE
BFD-243: Update Contributing and README instructions

### PR DESCRIPTION
**Pull request by cbrunefearless**
**JIRA Ticket:**
[BFD-243](https://jira.cms.gov/browse/BFD-243)

**User Story or Bug Summary:**
As a developer onboarding to the BFD, I want to get setup with as little outside help as possible and contribute my first issue. 

### What Does This PR Do?

- Breaks out Contributing guidance and Local Development Guidance
- Reconciles previous local development instructions into a single README
- Moves eclipse configuration assets to docs/assets

### What Should Reviewers Watch For?

If you're reviewing this PR, please check these things, in particular:

N/A

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then here's a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence: N/A.
    * Does this PR add any new software dependencies? **No**.
    * Does this PR modify or invalidate any of our security controls? **No**.
    * Does this PR store or transmit data that was not stored or transmitted before? **No**.
* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR should not be merged unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons? **No**.

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following pre-requisite changes have been fully deployed:

N/A

### Submitter Checklist

I have gone through and verified that...:

* [x] This PR has been refactored and rebased (for help, see: [this](https://dev.to/maxwell_dev/the-git-rebase-introduction-i-wish-id-had) and [this](https://raphaelfabeni.com/git-editing-commits-part-1/), if needed, so that it is as small as it can reasonably be, in order to:
    1. Ensure that any problems it causes have a small "blast radius".
    2. Ensure that it'll be easier to rollback if that becomes necessary.
    3. Ease the burden on reviewers.
* [ ] The commits in this PR:
    1. Reasonably explain the "what" and "why" of the changes.
    2. Leverage JIRA's [smart commits](https://confluence.atlassian.com/jirasoftwarecloud/processing-issues-with-smart-commits-788960027.html) to reference the JIRA ticket that they're associated with. For example, aim for commit messages like this (note also [the 50/72 formatting](https://stackoverflow.com/q/2290016) used here):
              
* [x] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [x] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [x] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [x] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. BFD, SLS, etc.).
* [x] Any deviations from the other policies in the [DASG Engineering Standards](../policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,84 +1,58 @@
...
 ### Contributing changes
 
 Contributions to the BFD are welcome from any party inside CMS.
 Small changes like "good first issues" can be submitted for consideration without any additional process.
 
-Any [substantive change must go though an RFC process](#proposing-substantive-changes) before work on the change itself can start.
+Any [substantive change must go though an RFC process](#proposing-substantive-changes) before work on the change itself can start. 
 
-**FAQ**
+Any code changes should be properly commented and accompanied by appropriate unit test as per DASG engineering guidelines.
```
Comment (jzulim) [inline CONTRIBUTING.md:27]:
```suggestion
Any code changes should be properly commented and accompanied by appropriate unit and integration test as per DASG engineering guidelines.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,84 +1,58 @@
...
 Substantive changes need to go through a design process involving the core team.
 Opening an RFC provides a path for this inclusion and process.
 Start an RFC by copying the file [`rfcs/0000-template.md`](rfcs/0000-template.md) to `rfcs/0000-<my idea>.md` and fill in the details. 
-Once filled out [submit a pull request](#opening-a-pr).
-The core team will respond with design feedback, and the author should be prepared to revise it in response.
+Open a PR using the RFC template [submit a pull request](#opening-a-pr).
 The RFC will remain open for a 2 week period, at the end of which a go/no-go meeting will be held.
-If approved by at least two core team members and there are no outstanding reservations it will be accepted, given a number, and merged into the master branch.
-Once accepted the author of the RFC and their team can scope the work within their regular process.
+If approved by at least two core team members and there are no outstanding reservations, the RFC will be accepted and given a number.
```
Comment (jzulim) [inline CONTRIBUTING.md:48]:
```suggestion
If approved by at least two core team members and there are no outstanding reservations, the RFC will be accepted.
```

The RFC number is selected when issuing the PR, as the next available.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -87,23 +61,67 @@ A: There is no strict definition however examples of substantive changes are:
...
+
+If you have any questions feel free to reach out on the [#bfd](https://cmsgov.slack.com/archives/C010WDXAZFZ) channel in CMS slack!
+
+**FAQ**
+
+Q: What if the core team rejects my PR?
+
+A: The BFD core team will commit to never rejecting a PR without providing a path forward. The developer who put up the PR should review any feedback, and discuss with their product manager the scope of the work that is now outstanding.
+
 Setting up the FHIR server locally
```
Comment (jzulim) [inline CONTRIBUTING.md:125]: I think we wanted to move this from the contributing and into the readme. Lines 125 through 288 need a new home, I believe.

Comment (egkaufer) [reply CONTRIBUTING.md:125]: I'm happy to move those to the README instead - @cbrunefearless @keithdadkins does that work?

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,84 +1,58 @@
+
 # Welcome!
 
 We want to ensure a welcoming environment for all of our projects. Our staff follow the [18F Code of Conduct](https://github.com/18F/code-of-conduct/blob/master/code-of-conduct.md) and all contributors should do the same.
 
+Although this is a public repo, contributing to the BFD is for CMS approved contributors only, not outside contributors.
```
Comment (karlmdavis) [inline CONTRIBUTING.md:6]:
```suggestion
Although this is a public repo, contributing to the BFD is for CMS-approved contributors only, not outside contributors.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,84 +1,58 @@
...
+### Background
 
-If you have any questions feel free to reach out on the [#bfd](https://cmsgov.slack.com/archives/C010WDXAZFZ) channel in CMS slack!
+BFD exists to enable the CMS Enterprise to drive innovation in data sharing so that beneficiaries and their healthcare partners have the data they need to make informed decisions about their healthcare.
 
-**FAQ**
+We provide a comprehensive, performant, and trustworthy platform to transform the way that the CMS enterprise shares and uses data by providing Coverage, Patient, and EOB FHIR-formatted data.
 
-Q: What if the core team rejects my PR?
+Review the [README](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/README.md) for additional information on the BFD.
```
Comment (karlmdavis) [inline CONTRIBUTING.md:17]:
```suggestion
Review the [README](https://github.com/CMSgov/beneficiary-fhir-data/blob/master/README.md) for additional information on BFD. 
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,84 +1,58 @@
...
-Please fill out each section of the body of the PR or quickly justify why the section does not apply to this particular change.
-Reviewers will automatically be suggested or requested based on the projects CODEOWNERS file, feel free to add other reviewers if desired.
-Once all automated checks pass and two reviewers have approved your pull request, a code owner from the core team will do a final review and make the decision to merge it in or not.
+### Background
 
-If you have any questions feel free to reach out on the [#bfd](https://cmsgov.slack.com/archives/C010WDXAZFZ) channel in CMS slack!
+BFD exists to enable the CMS Enterprise to drive innovation in data sharing so that beneficiaries and their healthcare partners have the data they need to make informed decisions about their healthcare.
 
-**FAQ**
+We provide a comprehensive, performant, and trustworthy platform to transform the way that the CMS enterprise shares and uses data by providing Coverage, Patient, and EOB FHIR-formatted data.
```
Comment (karlmdavis) [inline CONTRIBUTING.md:15]:
```suggestion
We provide a comprehensive, performant, and trustworthy platform to transform the way that the CMS enterprise shares and uses data by providing [FHIR](https://www.hl7.org/fhir/)-formatted data, including beneficiary demographic, enrollment, and claims data.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,84 +1,58 @@
...
 
-A: The BFD core team will commit to never rejecting a PR without providing a path forward. The developer who put up the PR should review any feedback, and discuss with their product manager the scope of the work that is now outstanding.
 
 ### Contributing changes
 
 Contributions to the BFD are welcome from any party inside CMS.
 Small changes like "good first issues" can be submitted for consideration without any additional process.
 
-Any [substantive change must go though an RFC process](#proposing-substantive-changes) before work on the change itself can start.
+Any [substantive change must go though an RFC process](#proposing-substantive-changes) before work on the change itself can start.
```
Comment (karlmdavis) [inline CONTRIBUTING.md:25]:
```suggestion
Any [substantive change must go though an RFC process](#proposing-substantive-changes) before work on the change itself can start.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,84 +1,58 @@
...
 Contributions to the BFD are welcome from any party inside CMS.
 Small changes like "good first issues" can be submitted for consideration without any additional process.
 
-Any [substantive change must go though an RFC process](#proposing-substantive-changes) before work on the change itself can start.
+Any [substantive change must go though an RFC process](#proposing-substantive-changes) before work on the change itself can start. 
 
-**FAQ**
-
-Q: How do I know what “first issues” are up for grabs? 
+Any code changes should be properly commented and accompanied by appropriate unit and integration test as per DASG engineering guidelines.
```
Comment (karlmdavis) [inline CONTRIBUTING.md:27]:
```suggestion
Any code changes should be properly commented and accompanied by appropriate unit and integration test as per [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md).
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -1,84 +1,58 @@
...
 
-A: First issues will be tracked on the BFD’s jira board and labeled (via versions) “Good First Issue”
+![Alt text](docs/assets/BFDContributionDiagram.png "Diagram")
```
Comment (karlmdavis) [inline CONTRIBUTING.md:29]:
```suggestion
![Workflow diagram for making a contribution to BFD](docs/assets/BFDContributionDiagram.png "Making a Contribution to BFD")
```

Comment (karlmdavis) [inline CONTRIBUTING.md:29]: @egkaufer Can you export this diagram as an SVG file, instead? That way, it'll be much easier for browsers and screen readers (for vision-impaired folks) to see.

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -1,84 +1,58 @@
...
+**FAQ**
 
 Q: What kind of changes don't require an RFC?
 
-A: In general bug fixes and small changes that do not affect behavior or meaning. If you're unsure please quickly ask in the [#bfd](https://cmsgov.slack.com/archives/C010WDXAZFZ) channel.
+A: In general bug fixes and small changes that do not affect behavior or meaning. If you're unsure please quickly ask in the [#bfd](https://cmsgov.slack.com/archives/C010WDXAZFZ) channel or in the Scrum of Scrums meeting. If your PR involves a substantial change, it will be rejected and you will be asked to go through the RFC process.
+
+Q: How do I know what “first issues” are up for grabs? 
+
+A: First issues will be tracked on the BFD’s jira board and labeled (via versions) “Good First Issue”
```
Comment (karlmdavis) [inline CONTRIBUTING.md:39]:
```suggestion
A: First issues are tracked on the BFD’s Jira board with the "Good_First_Issue" label: <https://jira.cms.gov/issues/?jql=project%20%3D%20BFD%20AND%20status%20%3D%20Open%20AND%20labels%20%3D%20Good_First_Issue>
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -87,184 +61,63 @@ A: There is no strict definition however examples of substantive changes are:
...
-
-To run the BFD locally in a way that will allow you and other systems to interact with it some modifications need to be made so that it serves on a consistent port. Caution: Since this changes the code in the repository (server-start.sh) please keep in mind not to commit these changes.
-
-These changes are contained in the file `contributing/patches/allow_local_port_config.patch` and can be applied with 
-
-```
-make servable
-```
-To undo the changes run `make unservable`.
+### Getting started
```
Comment (karlmdavis) [inline CONTRIBUTING.md:81]: I'd suggest replacing this entire section with a brief pointer to one of these:

* <https://github.com/CMSgov/beneficiary-fhir-data/blob/master/README.md>
* <https://github.com/CMSgov/beneficiary-fhir-data/blob/master/dev/devenv-readme.md>

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -87,184 +61,63 @@ A: There is no strict definition however examples of substantive changes are:
...
-      context: ../../bb.dev/msls
-      dockerfile: Dockerfile
-    command: msls
-    ports: 
-      - "8080:8080"
-```
+If you have any questions feel free to reach out on the [#bfd](https://cmsgov.slack.com/archives/C010WDXAZFZ) channel in CMS slack!
 
-Bringing these systems up together:
+**FAQ**
```
Comment (karlmdavis) [inline CONTRIBUTING.md:119]: Isn't this duplicative of the FAQ section above?

File: README.md [hunk 13]
```diff
@@ -1,11 +1,18 @@
 MyMedicare.gov BlueButton Parent POM
 ====================================
 
-This repo just contains a Maven parent POM and some other build/dev resources that are shared by the CMS/MyMedicare.gov Java projects.
+## About
```
Comment (karlmdavis) [inline README.md:4]: Please update the document title, above, as well:

```
# Beneficiary FHIR Data Server (BFD)

File: README.md [hunk 14]
```diff
@@ -23,3 +30,173 @@ This project is in the worldwide [public domain](LICENSE.md). As stated in [CONT
 > This project is in the public domain within the United States, and copyright and related rights in the work worldwide are waived through the [CC0 1.0 Universal public domain dedication](https://creativecommons.org/publicdomain/zero/1.0/).
 >
 > All contributions to this project will be released under the CC0 dedication. By submitting a pull request, you are agreeing to comply with this waiver of copyright interest.
+
```
[the remark this answers is not in the document]
Comment (egkaufer) [reply README.md:33]: @keithdadkins @cbrunefearless @jzulim Would any of you be able to take a look at these and update the README instructions?

File: README.md [hunk 15]
```diff
@@ -1,11 +1,18 @@
...
+### DASG Mission
+Drive innovation in data sharing so that beneficiaries and their healthcare partners have the data they need to make informed decisions about their healthcare. 
+
+### BFD Mission
+Enable the CMS Enterprise to drive innovation in data sharing so that beneficiaries and their healthcare partners have the data they need to make informed decisions about their healthcare. 
+
+### BFD Vision
+Provide a comprehensive, performant, and trustworthy platform to transform the way that the CMS enterprise shares and uses data. 
 
 ## Releases
```
[the remark this answers is not in the document]
Comment (egkaufer) [reply README.md:17]: Can do!

File: CONTRIBUTING.md (hunk 11 above)
Comment (egkaufer) [reply CONTRIBUTING.md:81]: @cbrunefearless @keithdadkins @jzulim any preference?

File: CONTRIBUTING.md (hunk 12 above)
Comment (egkaufer) [reply CONTRIBUTING.md:119]: It looks like the question about a rejected PR (lines 121-123) was moved here - should we move it somewhere else?

File: CONTRIBUTING.md (hunk 11 above)
Comment (keithdadkins) [reply CONTRIBUTING.md:81]: The norm is to have a "Getting started" section in the main README that points to the CONTRIBUTING docs, or if it's simple enough, having the entire "getting started" instructions in the README (this is not likely what we want to do). 

If we want to maintain both local (no docker) and docker dev environments, we should have links to "Getting started with Docker" and "Getting started locally without Docker (Windows only)". The latter would point to the existing docs which is hard coded for Windows development until if/when we create directions for setting up dev environments on MacOS, Linux, etc.

File: README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (keithdadkins) [reply README.md:24]: Not sure what we are talking about here? The license?

[the remark this answers is not in the document]
Comment (egkaufer) [reply README.md:24]: I removed the reference to the CONTRIBUTING doc under the License section, is that all that should be changed here?

[the remark this answers is not in the document]
Comment (karlmdavis) [reply README.md:24]: No, my comment refers to the content _below_ this line, which starts off with:

```
Setting up the FHIR server locally
---
Requirements: Docker 

Caution: Setting up your local environments requires git patches to run. Please make sure you `make unservable` and `make unloadadable` before you commit your changes to revert these patches. 

Let's begin!
...
```

File: CONTRIBUTING.md (hunk 11 above)
Comment (cbrunefearless) [reply CONTRIBUTING.md:81]: I concur with Keith on that..If the readme is simple enough, it should go in the readme, however more complex instructions should be outside of it.

Comment (jzulim) [reply CONTRIBUTING.md:81]: I've simplified the 'Getting Started' section with pointers to Docker and Windows Local instructions.

File: README.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (jzulim) [reply README.md:24]: I've reconciled the two instrucitons, in favor of the docker setup but with additional info from the original instructions such as AWS setup, Git Setup, and tacked on the bottom Eclipse specific instructions.

**Comment by egkaufer** [comment]
This is great, thank you all for your work on these docs!

**Pull request merged**